### PR TITLE
Changes to value comparison semantics

### PR DESCRIPTION
**Pull request by riyafa**

### CODE REVIEW

File: vxquery-core/src/main/java/org/apache/vxquery/runtime/functions/util/FunctionHelper.java [hunk 1]
```diff
@@ -487,316 +487,320 @@ public static boolean arraysEqual(byte[] bytes1, int offset1, int length1, byte[
...
-
-                case ValueTag.XS_NOTATION_TAG:
-                    tvp1.getValue(tp1.utf8sp);
-                    switch (tid2) {
-                        case ValueTag.XS_NOTATION_TAG:
-                            tvp2.getValue(tp2.utf8sp);
-                            return aOp.operateNotationNotation(tp1.utf8sp, tp2.utf8sp);
-                    }
-                    break;
+            if (tid1 == ValueTag.JS_NULL_TAG || tid2 == ValueTag.JS_NULL_TAG) {
```
Comment (prestoncarman) [inline vxquery-core/src/main/java/org/apache/vxquery/runtime/functions/util/FunctionHelper.java:495]: Please add this tag to the switch statement. The first switch is for the first argument and the second switch is for the second argument.

File: vxquery-xtest/src/test/resources/Queries/XQuery/Json/Functions/value_comparison.xq [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+   
+   Unless required by applicable law or agreed to in writing,
+   software distributed under the License is distributed on an
+   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+   KIND, either express or implied.  See the License for the
+   specific language governing permissions and limitations
+   under the License. :)
+
+(: JSONiq Changes to value comparison semantics :)
+1 eq null, null eq null, 1 ne null, null lt 1
```
Comment (prestoncarman) [inline vxquery-xtest/src/test/resources/Queries/XQuery/Json/Functions/value_comparison.xq]: Please add all possible comparison checks. (eq, ne, lt, gt, ge, le) and (null value, value null, and null null)

File: vxquery-xtest/src/test/resources/Queries/XQuery/Json/Functions/general_comparison.xq [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+   under the License. :)
+
+(: JSONiq Changes to value comparison semantics :)
+(null, 2) = (1, 3),
+(null, 2) != (1, 3),
+(null, 2) < (1, 3),
+(null, 2) <= (1, 3),
+(null, 1) > (1, 3),
+(null, 0) >= (1, 3),
+null < 1,
```
Comment (prestoncarman) [inline vxquery-xtest/src/test/resources/Queries/XQuery/Json/Functions/general_comparison.xq:25]: Please add equal and not equal.

File: vxquery-core/src/main/java/org/apache/vxquery/datamodel/accessors/AbstractSequencePointable.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.vxquery.datamodel.accessors;
+
+import org.apache.hyracks.data.std.api.AbstractPointable;
+import org.apache.hyracks.data.std.api.IPointable;
+import org.apache.hyracks.data.std.primitive.IntegerPointable;
+
+public class AbstractSequencePointable extends AbstractPointable {
```
Comment (westmann) [inline vxquery-core/src/main/java/org/apache/vxquery/datamodel/accessors/AbstractSequencePointable.java:23]: I'm confused. This class is already in master. Why is it in this change?

File: vxquery-core/src/main/java/org/apache/vxquery/datamodel/accessors/SequencePointable.java [hunk 5]
```diff
@@ -17,15 +17,11 @@
...
-import org.apache.hyracks.data.std.api.AbstractPointable;
 import org.apache.hyracks.data.std.api.IPointable;
 import org.apache.hyracks.data.std.api.IPointableFactory;
-import org.apache.hyracks.data.std.primitive.IntegerPointable;
 import org.apache.hyracks.data.std.primitive.VoidPointable;
 
-public class SequencePointable extends AbstractPointable {
-    private static final int ENTRY_COUNT_SIZE = IntegerPointable.TYPE_TRAITS.getFixedLength();
-    private static final int SLOT_SIZE = IntegerPointable.TYPE_TRAITS.getFixedLength();
+public class SequencePointable extends AbstractSequencePointable {
```
Comment (westmann) [inline vxquery-core/src/main/java/org/apache/vxquery/datamodel/accessors/SequencePointable.java:24]: This change should also be in master ...

**Pull request closed without merging**